data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -84,26 +91,39 @@ ReceiveMessageRequest createReceiveMessageRequest() {
     }
 
     private Map<Message, S3EventNotification.S3EventNotificationRecord> getS3MessageEventNotificationRecordMap(final List<Message> sqsMessages) {
-        return sqsMessages.stream().collect(Collectors.toMap(message -> message, this::convertS3EventMessages));
+        return sqsMessages.stream()
+                .collect(HashMap::new,
+                        (messageRecordMap, message) -> messageRecordMap.put(message, convertS3EventMessages(message)),
```

Comment:
Why can't we use `Collections.toMap` anymore?

Reply:
Collections.toMap doesn't allow null values.